### PR TITLE
SWIFT-1470 Fix failing load balancer tests

### PR DESCRIPTION
**Pull request by patrickfreed**
SWIFT-1470 / SWIFT-1319

### CODE REVIEW

File: Sources/TestsCommon/CommonTestUtils.swift [hunk 1]
```diff
@@ -188,7 +188,6 @@ public enum TestTopologyConfiguration: String, Decodable {
             helloReply["isreplicaset"] != true
         {
             self = .single
-            // TODO: SWIFT-1319: eventually, we should be able to just check for serviceId here.
```
Comment (patrickfreed) [inline Sources/TestsCommon/CommonTestUtils.swift:191]: as discussed offline, serviceId only shows up in the initial handshake, so we can't check for it here and need to still rely on the environment variable.

File: etc/vendor-libmongoc.sh [hunk 2]
```diff
@@ -153,8 +153,6 @@ echo "RENAMING header files"
 echo "PATCHING libmongoc"
 git apply ${ETC_DIR}/lower-minheartbeatfrequencyms.diff
 git apply ${ETC_DIR}/inttypes-non-modular-header-workaround.diff
-# TODO SWIFT-1319: Remove.
-git apply ${ETC_DIR}/expose-mock-service-id.diff
```
Comment (kmahar) [inline etc/vendor-libmongoc.sh:157]: should we delete this diff file now too?

**Review by kmahar — APPROVED** [review]
LGTM mod one comment

**Review by isabelatkinson — APPROVED** [review]
lgtm mod kaitlin's comment

**Pull request merged**